### ISSUE TITLE
Invalid response, status code: 500

### ISSUE THREAD

**Issue by TomS1985** [lostfields/python-panasonic-comfort-cloud#82]
Since today, 2:00am I don't get any response from the Comfort Cloud.
It seems like Panasonic worked on the system last night - after that I had these issues.

I got these errors this night:
from 1am to 3am this error:
`pcomfortcloud.session.ResponseError: Invalid response, status code: 503 - Data: {"message":"12/11/2023 01:00 ～ 12/11/2023 03:00 (UTC ± 00)","code":5300}`

after 3am the following error:
`pcomfortcloud.session.ResponseError: Invalid response, status code: 500 - Data: {"code":5001,"message":"DB system error due to db system"}`

after a few trys, it changed to (Panasonic blocked the account for 24 hours):
`pcomfortcloud.session.ResponseError: Invalid response, status code: 429 - Data: {"code":42901,"message":"Too Many Requests"}`

With a new account I get the following response:

```
Traceback (most recent call last):
  File "/usr/local/bin/pcomfortcloud", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/pcomfortcloud/__main__.py", line 202, in main
    session.login()
  File "/usr/local/lib/python3.11/dist-packages/pcomfortcloud/session.py", line 99, in login
    self._create_token()
  File "/usr/local/lib/python3.11/dist-packages/pcomfortcloud/session.py", line 135, in _create_token
    raise ResponseError(response.status_code, response.text)
pcomfortcloud.session.ResponseError: Invalid response, status code: 500 - Data: {"code":5001,"message":"DB system error due to db system"}
```

Is this problem already known? I'm executing pcomfortcloud automatically every 15 minutes with my home automation system (FHEM). If I execute pcomfortcloud as root user manually on the console, it works just fine.

Thank you and best regards,
Tom

**Comment by LudovicRousseau** [comment]
Exact same problem for me.

**Comment by mchacher** [comment]
Same here as well. I have been banned for 24 hours by Panasonic. I was a bit brutal asking for data every 15s. 
If I use panasonic application on the mobile app I have an error message telling me that a suspicious activity was detected on my account, and it has been blocked for 24 hours. They invite to dissociate my account to any non authorized application.

**Comment by lostfields** [comment]
Well, that doesn't look very promising. 

Looks like the have implemented a algorithm where they see the sequence of requests differs from their requests in their app.

I don't think this module is flooding with unnecessary requests unless you poll often.

**Comment by TomS1985** [comment]
I'm confused by the fact, that pcomfortcloud gets values from the Panasonic Cloud, if I run it manually on my linux shell (as a root user).
If its executed automatically by following fhem command, it gets the error code:
`system('pcomfortcloud -s true ' . $usr . ' ' . $pwd . ' get 1 > panasonicvalues.txt');`

So the python pcomfortcloud itself seems to work...?

**Comment by mchacher** [comment]
@TomS1985 can you give the example of the command that is working for you "manually"? Despite the app being functional now, nothing seems to be working on my end. The ban was lifted after 24 hours.

**Comment by TomS1985** [comment]
I executed the command "pcomfortcloud -s true ACCOUNT PASSWORD get 1" on my linux shell, which resulted in:

```
reading from device 'Garage' (40904d8388bee143de39ad46dfdcd4da7f62073c0aaece845f7e49f4d16adf19)
id                       : 40904d8388bee143de39ad46dfdcd4da7f62073c0aaece845f7e49f4d16adf19
parameters
    temperatureInside    : 7
    temperatureOutside   : 3
    temperature          : 8
    power                : Off
    mode                 : Heat
    fanSpeed             : High
    airSwingHorizontal   : Right
    airSwingVertical     : Up
    eco                  : Auto
    nanoe                : On
```

**Comment by mchacher** [comment]
thanks @TomS1985. It is not working on my end. 
`Traceback (most recent call last):
  File "/Users/mchacher/Documents/08_Python/Panasonic/python-panasonic-comfort-cloud/pcomfortcloud.py", line 5, in <module>
    __main__.main()
  File "/Users/mchacher/Documents/08_Python/Panasonic/python-panasonic-comfort-cloud/pcomfortcloud/__main__.py", line 202, in main
    session.login()
  File "/Users/mchacher/Documents/08_Python/Panasonic/python-panasonic-comfort-cloud/pcomfortcloud/session.py", line 99, in login
    self._create_token()
  File "/Users/mchacher/Documents/08_Python/Panasonic/python-panasonic-comfort-cloud/pcomfortcloud/session.py", line 135, in _create_token
    raise ResponseError(response.status_code, response.text)
pcomfortcloud.session.ResponseError: Invalid response, status code: 500 - Data: {"code":5001,"message":"DB system error due to db system"}`
Even the list command is rejected.

**Comment by mchacher** [comment]
@LudovicRousseau are you still facing the problem as well?

**Comment by msh321** [comment]
I have **_exactly_** the same problem: `Invalid response, status code: 500 - Data: {"code":5001,"message":"DB system error due to db system"}`.  I have tried `list` and `get` operations and I have tried from two different Linux servers, but in all cases I'm getting the `"code":5001` error.

I can see from my log files, that everything worked up until approx. 34 hours ago.  Then, all of a sudden, I got the `"code":42901` error, which lasted 24 hours.  For the last approx. 10 hours the error has changed to `"code":5001`.   😢 

_(I'm running `pcomfortcloud` version `0.0.24`.)_

**Comment by martinkaltoft** [comment]
I have the exact same symptoms:
./pcomfortcloud.py -r true -s true USER PASSWORD history 1 Day 20231212
--- creating token by authenticating
Traceback (most recent call last):
  File "pcomfortcloud.py", line 5, in <module>
    __main__.main()
  File "__main__.py", line 202, in main
    session.login()
  File "session.py", line 99, in login
    self._create_token()
  File "session.py", line 135, in _create_token
    raise ResponseError(response.status_code, response.text)
pcomfortcloud.session.ResponseError: Invalid response, status code: 500 - Data: {"code":5001,"message":"DB system error due to db system"}

I tried running the command as root with no change. 
I also tried changing 
"X-APP-VERSION": "1.19.0" 
to 
"X-APP-VERSION": "1.19.1" 
in session.py as that should be the current version, with no change in behaviour.

**Comment by msh321** [comment]
**Update:**
Now, the error has changed from `"code":500` to `"code":42901` (Too Many Requests) on my account.
I have run the `pcomfortcloud` command a few times -- perhaps 50 or so -- over the last few hours.  This seems to be really useless now...  What a shame.

**Comment by mkz212** [comment]
> **Update:** Now, the error has changed from `"code":500` to `"code":42901` (Too Many Requests) on my account. I have run the `pcomfortcloud` command a few times -- perhaps 50 or so -- over the last few hours. This seems to be really useless now... What a shame.

I have the same! Again a 24 h lock 😓

**Comment by majurgens** [comment]
Since it seems everyone is now tending towards what I noted earlier - I think we are all headed for non-functional code.

They appear to be targetting "non authorized applications"

If they are actively trying to block "non authorized applications" then this could be an endless cat and mouse game. Is there perhaps a way we can become "authorized"?

**Comment by LudovicRousseau** [comment]
For me the error is now:
```
pcomfortcloud.session.ResponseError: Invalid response, status code: 500 - Data: {"code":5001,"message":"DB system error due to db system"}
```
But I have no error if I use the iOS "Comfort Cloud" application (I had to accept new licenses).

So the error is not linked to my account, but to what python-panasonic-comfort-cloud is doing.
Maybe they filter by user-agent or some other header field?

**Comment by TomS1985** [comment]
How do you execute pcomfortcloud and which version do you use?

I've tested my pcomfortcloud again - it gets the correct values and I can set values (heater On/off) via console. Only by executing it on FHEM, I get the errors. But what's the difference between running it manually and running it by a "system" command on FHEM? If I run it as user "pi" (root) it, works fine, if I run it as user "fhem" (FHEM executes all the system-commands as user "fhem"), I get the error:
`Invalid response, status code: 500 - Data: {"code":5001,"message":"DB system error due to db system"}`

I'm using the latest version 0.0.24

Seems a problem of the user?! Can anyone confirm that?

**Comment by msh321** [comment]
I just tried the `list` command again -- first as a "normal" user (`msh`) and then as `root`.  In both cases I'm (still!) getting the error `Invalid response, status code: 500 - Data: {"code":5001,"message":"DB system error due to db system"}`.  😭

**Comment by Auka84** [comment]
Hey. I use another integration, this one https://github.com/sockless-coding/panasonic_cc. and we have same issue

so panasonic change a thing at their side

since yesterday, mine is back and is working. But i'm the only one and i don't know why it's working for me and not for all others 
https://github.com/sockless-coding/panasonic_cc/issues/147

**Comment by LudovicRousseau** [comment]
On a _new_ system I did:
```
$ git rev-parse  HEAD
80dbbfb22a4405e8908d32517a44b25bcf132779
$ ./pcomfortcloud.py ludovic.rousseau@xxxxxx xxxxxxxx list
Traceback (most recent call last):
  File "/home/rousseau/Documents/github/python-panasonic-comfort-cloud/./pcomfortcloud.py", line 5, in <module>
    __main__.main()
  File "/home/rousseau/Documents/github/python-panasonic-comfort-cloud/pcomfortcloud/__main__.py", line 202, in main
    session.login()
  File "/home/rousseau/Documents/github/python-panasonic-comfort-cloud/pcomfortcloud/session.py", line 99, in login
    self._create_token()
  File "/home/rousseau/Documents/github/python-panasonic-comfort-cloud/pcomfortcloud/session.py", line 135, in _create_token
    raise ResponseError(response.status_code, response.text)
pcomfortcloud.session.ResponseError: Invalid response, status code: 500 - Data: {"code":5001,"message":"DB system error due to db system"}
```
I tried (again) the iOS application and I have no problem to use this account.

**Comment by martinkaltoft** [comment]
They found the issue over at https://github.com/sockless-coding/panasonic_cc

I have created a pull request that solves the issue in this repository.

**Comment by TomS1985** [comment]
> They found the issue over at https://github.com/sockless-coding/panasonic_cc
> 
> I have created a pull request that solves the issue in this repository.

Thank you! Works fine again!

**Comment by msh321** [comment]
Cool, the fix works for me too.  😎 

**BTW:** Does anyone what the rate limit is?  I mean, how many requests can a user send per day, before the "rate-limit exceeded" error occurs.

**Comment by lostfields** [comment]
Great, thanks @martinkaltoft , released as [v0.0.25](https://github.com/lostfields/python-panasonic-comfort-cloud/releases/tag/v0.0.25)
